src/common/sidebar/containers/SidebarContainer.tsx

```diff
@@ -48,12 +49,34 @@ export function SidebarContainer({ searchQuery, term }: SidebarContainerProps):
     term: term as Term,
     queryParams: { in_session: filter },
   });
+  // re assign the type of subjects so inSession property is available
+  const inSessionSubjects: InSessionSubject[] | null = subjects;
 
   const loading = subjectsLoading || coursesLoading;
 
   // sorts the list of subjects alphabetically
-  const sortedSubjects = useMemo(() => subjects?.sort((a, b) => (a.subject > b.subject ? 1 : -1)), [subjects]);
+  let sortedSubjects = useMemo(() => inSessionSubjects?.sort((a, b) => (a.subject > b.subject ? 1 : -1)), [
+    inSessionSubjects,
+  ]);
   const parsedCourses = useMemo(() => computeParsedCourses(courses), [courses]);
+  //if filter is set then we want to add (Not offered) to subjects
+
+  if (filter) {
+    sortedSubjects = sortedSubjects?.filter((subject) => {
+      if (parsedCourses[subject.subject]) {
+        subject.inSession = true;
+        return subject;
+      } else {
+        subject.inSession = false;
+        return subject;
+      }
+    });
+  } else {
+    sortedSubjects = sortedSubjects?.filter((subject) => {
+      subject.inSession = true;
+      return subject;
+    });
+  }
```

Comment:
It would be better to add this `filter` logic within the `sortedSubjects` / `computeParsedCourses` as this avoids sidesteps the optimization of the `useMemo`.